File: dp/dp_a.py
"""
AtCoder Educational DP Contest
https://atcoder.jp/contests/dp/tasks/dp_a

A - Frog 1
"""
INF = 100000007

def cost(i, j):
    return abs(hs[i] - hs[j])

def solve(hs):
    dp = [INF] + [0] * n
    for i in range(2, n + 1):
        dp[i] = min(dp[i - 2] + cost(i - 2, i), dp[i - 1] + cost(i - 1, i))
    return dp[n]

n = int(input())
hs = [INF] + [int(s) for s in input().split()]
assert len(hs) == n + 1
res = solve(hs)
print(res)

File: dp/dp_a.cc
#include <iostream>
#include <vector>
using namespace std;

int N;
int h[100005];
long long dp[100005];

int main()
{
    cin >> N;
    for (int i = 1; i <= N; ++i) {
        cin >> h[i];
    }

    dp[1] = 0LL;
    for (int x = 2; x <= N; ++x) {
        dp[x] = dp[x - 1] + abs(h[x] - h[x - 1]);
        if (x >= 3)
            dp[x] = min(dp[x], dp[x - 2] + abs(h[x] - h[x - 2]));
    }

    cout << dp[N] << endl;
}